AppTaskQueue/appscale/taskqueue/statistics.py

```diff
@@ -1,92 +1,94 @@
-  def matches(self, request_info):
-    return request_info.api == REST_API
+def summarize_rest_request(request_info):
+  return request_info.api == REST_API
 
 
 # Define categorizers for grouping requests
-class PBMethodCategorizer(categorizers.Categorizer):
-  def category_of(self, req_info):
-    if req_info.api != PROTOBUFFER_API:
-      return categorizers.HIDDEN_CATEGORY
-    return req_info.pb_method
+def categorize_by_pb_method(request_info):
+  if request_info.api != PROTOBUFFER_API:
+    return stats_manager.HIDDEN_CATEGORY
+  return request_info.pb_method
+
+
+def categorize_by_rest_method(request_info):
+  if request_info.api != REST_API:
+    return stats_manager.HIDDEN_CATEGORY
+  return request_info.rest_method
+
+
+def categorize_by_pb_status(request_info):
+  if request_info.api != PROTOBUFFER_API:
+    return stats_manager.HIDDEN_CATEGORY
+  return request_info.pb_status
+
+
+def categorize_by_rest_status(request_info):
+  if request_info.api != REST_API:
+    return stats_manager.HIDDEN_CATEGORY
+  return request_info.rest_status
+
 
+# define metrics
+def count_failed_requests(requests):
+  return sum(1 for request in requests
+             if request.pb_status != "OK" and request.rest_status != 200)
 
-class RestMethodCategorizer(categorizers.Categorizer):
-  def category_of(self, req_info):
-    if req_info.api != REST_API:
-      return categorizers.HIDDEN_CATEGORY
-    return req_info.rest_method
 
+def count_protobuff_requests(requests):
+  return sum(1 for request in requests if request.api == PROTOBUFFER_API)
 
-class PBStatusCategorizer(categorizers.Categorizer):
-  def category_of(self, req_info):
-    if req_info.api != PROTOBUFFER_API:
-      return categorizers.HIDDEN_CATEGORY
-    return req_info.pb_status
 
-class RestStatusCategorizer(categorizers.Categorizer):
-  def category_of(self, req_info):
-    if req_info.api != REST_API:
-      return categorizers.HIDDEN_CATEGORY
-    return req_info.rest_status
+def count_rest_requests(requests):
+  return sum(1 for request in requests if request.api == REST_API)
 
 
 # Instantiate request matchers
-FAILED_REQUEST = FailedRequestMatcher()
-PROTOBUFF_REQUEST = ProtobufferRequestMatcher()
-REST_REQUEST = RestRequestMatcher()
+FAILED_REQUEST = summarize_failed_request
+PROTOBUFF_REQUEST = summarize_protobuffer_request
+REST_REQUEST = summarize_rest_request
 
 # Instantiate request categorizers
-PB_METHOD_CATEGORIZER = PBMethodCategorizer(
-  categorizer_name="by_pb_method")
-REST_METHOD_CATEGORIZER = RestMethodCategorizer(
-  categorizer_name="by_rest_method")
-PB_STATUS_CATEGORIZER = PBStatusCategorizer(
-  categorizer_name="by_pb_status")
-REST_STATUS_CATEGORIZER = RestStatusCategorizer(
-  categorizer_name="by_rest_status")
+PB_METHOD_CATEGORIZER = categorize_by_pb_method
+REST_METHOD_CATEGORIZER = categorize_by_rest_method
+PB_STATUS_CATEGORIZER = categorize_by_pb_status
+REST_STATUS_CATEGORIZER = categorize_by_rest_status
```

Comment:
The comment is outdated, there's no instantiation any more.
Even more, I think there's no need for these constants any more and you can write function names directly in configuration dict.